File: Indusmática/5to ciclo/Progra 2/Lab 7 - Herencia/2024-2/Lab07_2024_2_Benchele/Bebida.cpp
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/cppFiles/class.cc to edit this template
 */

/* 
 * File:   Bebida.cpp
 * Author: Lenovo
 * 
 * Created on 12 de junio de 2025, 05:20 PM
 */

#include <iostream>
#include <iomanip>
#include <cstring>
#include "Bebida.h"
using namespace std;

Bebida::Bebida() {
    tamano = nullptr;
}

Bebida::~Bebida() {
    if(tamano != nullptr)delete tamano;
}

void Bebida::SetTamano(const char* tam) {
    if(tamano != nullptr)delete tamano;
    tamano = new char[strlen(tam) + 1];
    strcpy(tamano, tam);
}

void Bebida::GetTamano(char* tam) const {
    if(tamano == nullptr)tam[0] = 0;
    else strcpy(tam, tamano);
}

/*
 161,B,Agua Mineral,2.5,mediano
 */

//Leer datos
void Bebida::leerDatos(ifstream &arch){
    char aux[60];
    Producto::leerDatos(arch);
    arch>>aux;
    arch.get();
    SetTamano(aux);
}

//Imprimir
void Bebida::imprimir(ofstream &arch){
    Producto::imprimir(arch);
    arch<<"TAMAÑO: ";
    arch<<setw(20)<<left<<tamano<<endl;
}

File: Indusmática/5to ciclo/Progra 2/Lab10/2024-2/Lab10_2024_2-Santiago/src/lab10_2024_2/santiago/Bebida.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package lab10_2024_2.santiago;

import java.util.Scanner;

/**
 *
 * @author santi
 */
public class Bebida extends Producto{
    
    private char tipoBebida;

    /**
     * @return the tipoBebida
     */
    public char getTipoBebida() {
        return tipoBebida;
    }

    /**
     * @param tipoBebida the tipoBebida to set
     */
    public void setTipoBebida(char tipoBebida) {
        this.tipoBebida = tipoBebida;
    }

    @Override
    public void cargar(Scanner arch) {
        super.cargar(arch);
        tipoBebida = arch.next().charAt(0);
    }

    @Override
    public void imprimir() {
        super.imprimir();
        System.out.println("Tipo: " + tipoBebida);
    }
}
